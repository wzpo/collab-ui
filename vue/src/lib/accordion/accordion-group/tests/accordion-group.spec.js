import { mount, shallowMount } from '@vue/test-utils';
import AccordionGroup from '../index.vue';
import AccordionContent from '../../accordion-content/index.vue';
import AccordionHeader from '../../accordion-header/index.vue';
import Vue from 'vue';

Vue.component(AccordionContent.name, AccordionContent);
Vue.component(AccordionHeader.name, AccordionHeader);

describe('AccordionGroup', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(AccordionGroup, {
      slots: {
        default: `<md-accordion-header /><md-accordion-content />`
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render children', () => {
    const container = mount(AccordionGroup, {
      propsData: {
        isExpanded: true
      },
      slots: {
        default: `<md-accordion-header /><md-accordion-content />`
      }
    });

    expect(container.classes('md-accordion__group--active')).toEqual(true);
    expect(container.findAll('.md-accordion__header').length).toEqual(1);
    expect(container.findAll('.md-accordion__content').length).toEqual(1);
    expect(container.vm.isExpanded).toEqual(true);
  });

  it('should pass class attr', () => {
    const container = shallowMount(AccordionGroup, {
      attrs: {
        class: 'testClass'
      },
      slots: {
        default: `<md-accordion-header /><md-accordion-content />`
      }
    });

    expect(container.classes('testClass')).toEqual(true);
  });

  it('should pass disabled, focus and showSeparator props to children', () => {
    const container = mount(AccordionGroup, {
      propsData: {
        disabled: true,
        focus: true,
        showSeparator: false,
      },
      slots: {
        default: `<md-accordion-header /><md-accordion-content />`
      }
    });

    expect(container.classes('md-accordion__group--disabled')).toEqual(true);
    expect(container.find(AccordionHeader).props().disabled).toEqual(true);
    expect(container.find(AccordionHeader).props().focus).toEqual(true);
    expect(container.find(AccordionHeader).props().showSeparator).toEqual(false);
  });

});

