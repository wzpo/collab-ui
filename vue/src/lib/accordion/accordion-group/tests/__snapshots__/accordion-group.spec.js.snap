// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionGroup should match snapshot 1`] = `
<div class="md-accordion__group">
  <div role="button" tabindex="0" class="md-accordion__header"> <span class="md-accordion__header-icon"></span></div>
  <div class="md-accordion__content"></div>
</div>
`;
